src/inputs/SelectField.test.tsx

```diff
@@ -216,7 +216,102 @@ describe("SelectFieldTest", () => {
     expect(r.age()).toHaveValue("One");
   });
 
-  const options = [
+  it("can define and select 'unsetLabel' when options are an array", async () => {
+    // Given a Select Field with options that already available
+    const r = await render(
+      <TestSelectField
+        label="Age"
+        value="1"
+        unsetLabel="None"
+        options={labelValueOptions}
+        getOptionLabel={(o) => o.label}
+        getOptionValue={(o) => o.value}
+      />,
+    );
+    // When we focus the field to open the menu
+    r.age().focus();
+    // And we select the 'unset' option
+    click(r.getByRole("option", { name: "None" }));
+    // Then onSelect was called
+    expect(onSelect).toHaveBeenCalledWith(undefined);
```

Comment:
Hmmm. I think I agree with `opt: 0 | undefined`, but `V extends Value`, and Value is:
```
export type Value = string | number | null | undefined | boolean;
```
This already includes `undefined`, which I think should be okay, right? And actually would `opt` ever be `undefined`? the `opt` in this case would end up being the `const unsetOption = {};`, right?
I might be missing something.. but I think this is fine. I'll pull the build to Blueprint locally and do a type-check to see what happens.

Reply:
@bdow `V extends Value` means it _could_ be `string | number | null | undefined | boolean`; but it could also be _just_ `string`.

Reply:
I get what you're saying. Sorry it took me a bit to come around. It is less about what `V` could be, but what `V` is instantiated as by the caller. 10-4. Will update.
I will fix it, but fwiw, `type-check` didn't find any issues in `internal-frontend`. 

So do you think in the `onSelect` we'd do something more like:
```
onSelect={(values, options) => {
  if (values.length > 0 && options.length > 0) {
     const option = options[0];
     onSelect(values[0], option === unsetOption ? undefined : option);
  }
}}
```
Because `values[0]` will be `undefined`, but `options[0]` could be an empty object. So do you think we should return `undefined` or would an empty object be okay in this case?

Reply:
> I will fix it, but fwiw, type-check didn't find any issues in internal-frontend.

I was initially really surprised, but I guess in retrospect we probably ~95% of the time use GraphQL inputs where are defacto `someFoo: string | null | undefined`...

> So do you think in the onSelect we'd do something more like:

Right, I think that makes sense, as well at update the TypeScript definition of `onSelect` to be `value: V | undefined`; that is really what I'm nudging for, is that we don't we don't want the client to think "I'm using `SelectField<string>`" and suddenly get `value: undefined` passed to their `onSelect` method.

If all of our blueprint callers are already great with `V | undefined`, that's great, will make it easy to roll out then...

> I really want to break out the logic of the multi select and single select

Nice! +1